### ISSUE TITLE
Scheduling multiple notifications

### ISSUE THREAD

**Issue by Dargeth** [katzer/cordova-plugin-local-notifications#688]
Building an application for iOS(9) and Android(4.4.2) phone.
For our application we want to schedule 3 notifications.
Tried multiple ways, with different results (code below).

Tested with the first triggering after a minute and a 5 and 10 minute interval for second and third.
Longer durations might require the device setting not to freeze the application in background.

Option A, 
Android:
App in background:
First, second and third all trigger.
App killed:
First triggers, rest does not.

iOS:
App in background:
First triggers, rest does not.
App killed:
First triggers, rest does not.

Option B:
Results in 3 notification entries.

Android:
App in background:
First, second and third all trigger.
App killed:
First, second and third all trigger.

iOS:
App in background:
Third triggers, rest does not.
App killed:
Third triggers, rest does not.

Any ideas on how I could could make them all trigger, with a preference to having only one notification entry on the device.

Code:
Option A:

                cordova.plugins.notification.local.schedule({
                    id: 656,
                    title: 'title',
                    text: 'message',
                    at: timeFirstMessage
                });
                cordova.plugins.notification.local.on("trigger", function (notification) {
                    if (notification.id != 656)
                        return;

                    setTimeout(function () {
                        cordova.plugins.notification.local.update({
                            id: 656,
                            title: "title",
                            text: "message"
                        });
                    }, 300000);
                });
                cordova.plugins.notification.local.on("trigger", function (notification) {
                    if (notification.id != 656)
                        return;

                    setTimeout(function () {
                        cordova.plugins.notification.local.update({
                            id: 656,
                            title: "title",
                            text: 'message'
                        });
                    }, 600000);
                });

Option B:

                cordova.plugins.notification.local.schedule({
                    id: 656,
                    title: 'title',
                    text: 'message'
                    at: timeFirstMessage
                });

                cordova.plugins.notification.local.schedule({
                    id: 657,
                    title: 'title',
                    text: 'message',
                    at: timeSecondMessage
                });

                cordova.plugins.notification.local.schedule({
                    id: 658,
                    title: 'title',
                    text: 'message',
                    at: timeThirdMessage
                });

**Comment by Tawpie** [comment]
@Dargeth:

In option A, I would try adding an “at” to the update. The plugin may be simply updating the title and message without changing the fire schedule (and the scheduled fire time has already passed).

For option B, I’ve found that you can’t bombard the plugin with several individual schedule requests. Try putting the three into an array and using scheduleAll.

I am REAL curious though what happens if you tap on the notification in the notification bar after you kill (swipe kill) your app. Our app will launch just fine, but the ‘click’ event isn’t posted so we don’t know that the user clicked on a notification. We’ve worked around it, but!

**Comment by Dargeth** [comment]
In option A the trigger effect should go off once the first notification fires. 
So when the first notification comes up the other 2 are set simultaneously.
But for this to happen the application has to be running in the background, which is not an ideal situation.

For option B, putting them in to an array and then schedule all seems to work on iOS. But it needs some additional tweaking else the application will crash.

For notifications on iOS update the file src/ios/APPLocalNotificationOptions.m replace: -@property(readwrite, assign) NSDictionary* dict; with:
@property(nonatomic, retain) NSDictionary* dict;
* If you get a build error on iOS:
ld: 1 duplicate symbol for architecture x86_64 clang: error: linker command failed with exit code 1 (use -v to see invocation)
FIX: commenting out line 28 in platforms/ios/MyApp/Plugins/de.appplant.cordova.common.registerusernotificationsettings/AppDelegate+APPRegisterUserNotificationSettings.m
NSString* const UIApplicationRegisterUserNotificationSettings = @"UIApplicationRegisterUserNotificationSettings";

If the application is killed from the background and a notification is clicked, the application starts and picks up the .click parameter that was added. 
But in this case the only thing happening is saving which event was clicked to open the application.

**Comment by Dargeth** [comment]
And an old problem seems to have resurfaced.

If the device is locked longer then 15 minutes the events won't fire unless the device is manually turned on. Which then fires all the notifications that should have been there already.

In the issues overview this seems fixed with the 0.8 rebuild but I suppose something changed.

In the notification.java line 177 the alarm manager gets the property RTC_WAKEUP which should wake up the device and fire the event. Seems this is not the case.
Any advice on this is welcome. Resorting to keeping the entire phone awake 24/7 is really battery unfriendly.

**Comment by louloox** [comment]
Hi!
@Dargeth: Thanks man! I'm having hard times with the notification which doesn't trigger events when fired (after a long time locked)! And I've just seen your tip : Adding: uses-permission android:name="android.permission.WAKE_LOCK"
I'm trying that now, and hopefully it will do the job =)
Louloox

**Comment by katzer** [comment]
@louloox @Dargeth #766 sounds kind of related

**Comment by hammadjamil** [comment]
After spending lot of time on finding solution of this issue i haven't found any of the suggestion worked for me.
But finally I have find solution and root cause of the issue on my own
The issue is events like click and trigger doesn't work after the app kill because when app is opened after kill events are not bind.
So the solution of this issue is to rebind the events at the start of the app your problem will be solved
like i am working on ionic app i have put below code in app.js for notification click event
`$ionicPlatform.ready(function () {
cordova.plugins.notification.local.on("click", function (notification) {
                        cordova.plugins.notification.local.cancel(notification.id, function () {
                            // Notification was cancelled
                            console.log('notification is cancelled  : '+ notification.id);
                             
                        }, '');

                    });
 });`

Now events are bind and work properly even after app kill. i have tested it for android and hopefully work for ios as well

**Comment by louloox** [comment]
@hammadjamil 
Hi! Thanks for coming back on this subject. I completely postponed this issue as far as i'm concerned... 
But looking at your message put me back in :D 
Can you explain what was your issue please?
In my code, I was already rebinding my events on start : 

`var app = angular.module('myApp', ['ionic', 'myApp.controllers', 'myApp.services', 'myApp.services.sqlite'])

.run(function($ionicPlatform, ConfigurationSvc, $cordovaSQLite) {
  $ionicPlatform.ready(function() {

cordova.plugins.notification.local.on("schedule", function(notification) {
        console.log("scheduled: " + notification.id);
    });

    cordova.plugins.notification.local.on('trigger', function (notification) {    
          
        if(notification.id == notifIdSilentNotif)
        { 
          navigator.vibrate([500, 200, 500]);
....
....
`
But when a notification is triggered after a while, my phone doesn't vibrate, so I assume my events are unboud, and won't rebound before starting the app again (basically, when i click on a notif, my app starts and my phone vibrates as it should have before).

How is it working for you ? What does you notif when triggered ? And what does your "onclick" ?

Btw, i haven't updated the plugin for a while (10/2015), so maybe it has been fixed in the meantime =)

I'll try an update to see.

Cheers!
